src/server/main_service.cc

```diff
@@ -1010,6 +1010,59 @@ bool Service::InvokeCmd(const CommandId* cid, CmdArgList tail_args, ConnectionCo
   return true;
 }
 
+void Service::DispatchManyCommands(absl::Span<CmdArgList> args_list,
+                                   facade::ConnectionContext* cntx) {
+  ConnectionContext* dfly_cntx = static_cast<ConnectionContext*>(cntx);
+  DCHECK(!dfly_cntx->conn_state.exec_info.IsRunning());
+
+  auto* exec_cid = registry_.Find("EXEC");
+  vector<StoredCmd> stored_cmds;
+  intrusive_ptr<Transaction> dist_trans;
+
+  auto perform_squash = [&] {
+    if (stored_cmds.empty())
+      return;
+
+    if (!dist_trans) {
+      dist_trans.reset(new Transaction{exec_cid});
+      dist_trans->StartMultiNonAtomic();
+    }
```

Comment:
The resource initialization is a little ugly here, but I'm trying to remove any overhead in case squashing is not used. This is the most likely case for multi/exec transactions, which arrive in pipelines, but can only be executed once collected (so squashing doesn't make any sense)